core/src/main/scala/api/sfs.scala

```diff
@@ -2,6 +2,11 @@ package sfs
 package api
 
 trait Filesystem {
+
+  /** Allows for effects
+   */
+  type M[_]
```

Comment:
Maybe we should have a different name for this. On the other hand, `M[_]` clearly communicates no assumptions are made about.